src/main/java/com/routeme/api/UserController.java

```diff
@@ -40,15 +40,17 @@ public UserController(UserService service) {
 
     @RequestMapping(method = RequestMethod.POST, consumes = "application/json", produces = "application/json")
     @ResponseStatus(HttpStatus.CREATED)
-    UserDTO create(@RequestBody @Valid UserDTO userEntry) {
-        UserDTO userRecord = service.create(userEntry);
-        String userId = userRecord.getEmail();
-        predictionIOClient.addUserToClient(userId);
-        // SHOULD BE BATCHED
-        predictionIOClient.preferRouteType(userId, PredictionIOClient.PREFERENCE_ROUTE_TYPE_LEASTTIME);
-        predictionIOClient.preferRouteMode(userId, PredictionIOClient.PREFERENCE_ROUTE_MODE_BUS);
-        predictionIOClient.preferRouteMode(userId, PredictionIOClient.PREFERENCE_ROUTE_MODE_BUS);
-        predictionIOClient.preferRouteMode(userId, PredictionIOClient.PREFERENCE_ROUTE_MODE_UBAHN);
+    UserDTO signup(@RequestBody @Valid UserDTO userEntry) {
+        UserDTO userRecord = service.create(userEntry).orElseThrow(
+                () -> new UserAlreadyExistsException(userEntry.getEmail()));
+        return userRecord;
+    }
+
+    @RequestMapping(value = "/login", method = RequestMethod.POST, consumes = "application/json", produces = "application/json")
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://www.codacy.com/app/heshamhamdymassoud/RouteMe-API/file/2638676964/issues/source?bid=3287637&fileBranchId=3320857#l49)